docs/_posts/2020-01-29-boltwash.md

```diff
@@ -0,0 +1,65 @@
+---
+title: "Bolt + Wash"
+description: "Navigate your Bolt inventory in Wash"
+author: michaelsmith
+---
+
+One of the goals of Wash is to make remote infrastructure feel like a natural extension of your local environment. So while you can use [sshfs](https://github.com/libfuse/sshfs) to access remote filesystems over SSH, you still have to recall or lookup the name or IP address of the machine and mount it. After doing that you might still have to go elsewhere to lookup different properties of that machine (like what tags it has, what OS it's running, or what datacenter it's a part of). And if you have lots of machines you want to access frequently, you'll need some sort of organization to keep track of them. Wash uses different infrastructure APIs (such as AWS and GCP) to dynamically setup all of that.
+
+If you're not using a cloud provider but you are using [Puppet Bolt](https://puppetlabs.github.io/bolt), then you've already created groups of targets, assigned properties to those targets, and configured how to connect to them. We created the [boltwash](https://github.com/puppetlabs/boltwash) plugin to leverage your existing Bolt inventory in Wash.
+
+The Bolt plugin for Wash presents a view of Bolt targets organized by groups, and allows you to navigate their filesystems (currently only with SSH)
```

Comment:
I think this narrative is good enough for the Boltwash repo since people will have likely visited that repo after having some idea of what Boltwash does, but it should be further fleshed out for a Wash blog post/possible Puppet blog post where people have no idea what Boltwash does or what Wash is. Whatever narrative that's here should show off as much of the plugin as possible and only leave implicit details if we can assume that 99% of readers will be able to connect the dots and have that "Aha" moment.

As an example, I'm thinking of the following narrative

"If you're using Bolt, then we created the Boltwash plugin that lets you navigate your inventory... Let's check it out"

_Paste stree output here_

As you can see, the Bolt plugin organizes Bolt targets as groups, blah blah blah

Let's try ls'ing it.

```
ls bolt
```

(We can see the groups)

Let's cd into a group and ls its targets

```
cd group ls target
```

Now let's cd into a target and navigate its filesystem

```
<cd target, ls>
<paste output showing you ls'ing and cd'ing through things, cat'ing/tail'ing logs>
```

Cool, right? Now what's really great about Boltwash is you can use it to debug any failures. Here's an example

```
<Walkthrough your narrative here then link to some other interesting Boltwash use-cases>
```

I think embedding a screencast in the blog post that summarizes all of this stuff might be useful so people can see everything we just talked about.

Also may be worth mentioning the configurable Boltdir stuff and show off some target metadata (and possible filtering with `find` if the current metadata's useful for that).

I think Boltwash is an interesting enough idea that we could get away with a somewhat longer Blog post.

Reply:
This is good feedback. It'll take me a bit to work through it.